nsqadmin/static/js/views/lookup.js

```diff
@@ -20,14 +21,23 @@ var LookupView = BaseView.extend({
 
     initialize: function() {
         BaseView.prototype.initialize.apply(this, arguments);
-        $.ajax(AppState.url('/topics?inactive=true')).done(function(data) {
-            this.template = require('./lookup.hbs');
-            this.render({
-                'topics': _.map(data['topics'], function(v, k) {
-                    return {'name': k, 'channels': v};
-                })
-            });
-        }.bind(this));
+        $.ajax(AppState.url('/topics?inactive=true'))
+            .done(function(data) {
+                    this.template = require('./lookup.hbs');
```

Comment:
Swapping out templates like this seems a little strange to me (and prone to bugs). I usually have one template for a View. And if I need to insert spinners or error messages, I do that "manually". Typically, this ajax code would live in a model which would fire `Pubsub` events, like `loading`, and `error`. Views would then listen to these events and throw in error messages or spinners as needed.

All of this might seem to be more work than you want to do right now, however - especially if you're wanting to a rewrite (sayyyy... in React?)